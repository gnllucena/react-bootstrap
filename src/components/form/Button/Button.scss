.ant-btn-primary:focus, 
.ant-btn-primary:hover, 
.ant-btn:focus, 
.ant-btn:hover {
  color: unset !important;
  background-image: unset !important;
  background-color: unset !important;
  border-color: unset !important;
}

.btn {
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 3px;
  width: 100%;
  font-weight: 700 !important;

  &.btn-small {
    height: 38px;
    font-size: 15px !important;
  }

  &.btn-big {
    height: 57px;
    font-size: 17px !important;
  }

  &.btn-primary {
    color: $button-text !important;
    border-color: $primary !important;
    background-color: $primary !important;
  
    &:hover {
      color: $button-text-hover !important;
      border-color: $primary-hover !important;
      background-color: $primary-hover !important;
    }
  }

  &.btn-link {
    color: $primary !important;

    &:hover {
      color: $primary-hover !important;
      border: none;
    }
  }

  &.btn-default, .ant-btn-default:active {
    background-color: $button-background !important;
    border: 1px solid $button-background !important;
    color: $text !important;
    font-weight: 500 !important;
    
    &:hover {
      color: $text !important;
      background-color: transparent !important;
      border: 1px solid $button-background !important;
    }
  }

  svg {
    width: 18px;
    height: 18px;
    margin-right: 2px;
  }

  &::after {
    content: none;
  }
}