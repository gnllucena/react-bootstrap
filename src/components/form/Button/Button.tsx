import React, { Component, CSSProperties, FunctionComponent } from 'react';
import { Button as ButtonAntd } from 'antd';

import './Button.scss';

export interface ButtonProps {
  text: string,
  size: 'small' | 'big'
  type: 'submit' | 'button' | 'reset',
  design: 'primary' | 'default' | 'link' | 'ghost' | 'dashed' | 'text',
  icon?: Component
  loading?: boolean,
  disabled?: boolean,
  style?: CSSProperties,
  onClick?(): void
}

export const Button: FunctionComponent<ButtonProps> = ({
  text,
  size,
  type,
  design,
  icon,
  loading,
  disabled,
  style,
  onClick
}) => (
  <ButtonAntd
    className={`btn btn-${size} btn-${design}`}
    type={design}
    style={style}
    htmlType={type}
    icon={icon}
    loading={loading}
    disabled={loading ? true : disabled}
    onClick={() => {
      if (onClick) {
        onClick();
      }
    }}
  >
    {text}
  </ButtonAntd>
);

export default Button;
