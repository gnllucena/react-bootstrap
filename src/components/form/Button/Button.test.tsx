import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Button, ButtonProps } from './Button';

const parameters: ButtonProps = {
  text: 'button',
  type: 'submit',
  design: 'primary',
  size: 'small'
}

const component = (
  <Button {...parameters}></Button>
);

describe('Button', () => {
  test('Rendering', () => {
    const wrapper = renderer.create(component);

    let json = wrapper.toJSON();

    expect(json).toMatchSnapshot();
  });
});