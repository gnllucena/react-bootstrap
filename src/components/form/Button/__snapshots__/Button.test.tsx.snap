// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering 1`] = `
<button
  className="ant-btn ant-btn-primary btn btn-small btn-primary"
  onClick={[Function]}
  type="submit"
>
  <span>
    button
  </span>
</button>
`;
